omnigibson/prims/rigid_prim.py

```diff
@@ -279,44 +282,44 @@ def set_linear_velocity(self, velocity):
         Sets the linear velocity of the prim in stage.
 
         Args:
-            velocity (np.ndarray): linear velocity to set the rigid prim to. Shape (3,).
+            velocity (th.tensor): linear velocity to set the rigid prim to. Shape (3,).
         """
         self._rigid_prim_view.set_linear_velocities(velocity[None, :])
 
     def get_linear_velocity(self):
         """
         Returns:
-            np.ndarray: current linear velocity of the the rigid prim. Shape (3,).
+            th.tensor: current linear velocity of the the rigid prim. Shape (3,).
         """
-        return self._rigid_prim_view.get_linear_velocities()[0]
+        return th.tensor(self._rigid_prim_view.get_linear_velocities()[0])
 
     def set_angular_velocity(self, velocity):
         """
         Sets the angular velocity of the prim in stage.
 
         Args:
-            velocity (np.ndarray): angular velocity to set the rigid prim to. Shape (3,).
+            velocity (th.tensor): angular velocity to set the rigid prim to. Shape (3,).
         """
         self._rigid_prim_view.set_angular_velocities(velocity[None, :])
 
     def get_angular_velocity(self):
         """
         Returns:
-            np.ndarray: current angular velocity of the the rigid prim. Shape (3,).
+            th.tensor: current angular velocity of the the rigid prim. Shape (3,).
         """
-        return self._rigid_prim_view.get_angular_velocities()[0]
+        return th.tensor(self._rigid_prim_view.get_angular_velocities()[0])
 
     def set_position_orientation(self, position=None, orientation=None):
         # Invalidate kinematic-only object pose caches when new pose is set
         if self.kinematic_only:
             self.clear_kinematic_only_cache()
         if position is not None:
-            position = np.asarray(position)[None, :]
+            position = th.asarray(position)[None, :]
         if orientation is not None:
-            assert np.isclose(
-                np.linalg.norm(orientation), 1, atol=1e-3
+            assert math.isclose(
```

Comment:
another general comment: what is our general rule of using `math` functions rather than `torch` functions? In this case, maybe we should use `torch`? But in general, maybe we should quickly go over all the instance of `math` functions and confirm we intend to use `math`, not `torch`.